pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -452,4 +492,46 @@ static void _session_to_ep(const session_t *session, sock_udp_ep_t *ep)
     memcpy(&ep->addr.ipv6, &session->addr, sizeof(ipv6_addr_t));
 }
 
+#ifdef SOCK_HAS_ASYNC
+void _udp_cb(sock_udp_t *sock, sock_async_flags_t flags)
+{
+    if (flags & SOCK_ASYNC_MSG_RECV) {
+        sock_dtls_session_t remote;
+        ssize_t res = sock_udp_recv(sock, data, max_len, timeout,
+                                    &remote.ep);
+        if (res <= 0) {
+            DEBUG("sock_dtls: error receiving UDP packet: %d\n", (int)res);
+            return res;
+        }
+
+        _ep_to_session(&remote.ep, &remote.dtls_session);
+        /* TODO how to get dtls_sock here??! */
+        res = dtls_handle_message(dtls_sock->dtls_ctx, &remote->dtls_session,
+                                  (uint8_t *)data, res);
```

Comment:
Currently, there is no way to receive the `dtls_sock` here. Maybe some additional `void *cb_arg` parameter for the callbacks would be good. What do you think @kaspar030

Reply:
See https://github.com/RIOT-OS/RIOT/pull/13615